src/gafaelfawr/providers/oidc.py

```diff
@@ -199,6 +211,51 @@ async def logout(self, session: State) -> None:
         """
         pass
 
+    async def _get_ldap_uidNumber(self, uid: str) -> int:
+        """Retrieve the uid number from ldap
+
+        Parameters
+        ----------
+        uid : `str`
+            Username of the user.
+
+        Returns
+        -------
+        uidNumber : `int`
+            The numeric id of the user from LDAP.
+
+        Raises
+        ------
+        gafaelfawr.exceptions.LDAPException
+            The lookup using the uid_number_attr against the LDAP server was not
+            valid (attribute not in LDAP or resultant value not an integer).
+        """
+        assert self._ldap_config.uid_number_attr
+        attr = self._ldap_config.uid_number_attr
+        search = f"(&(Uid={uid}))"
```

Comment:
Does this still work if you use `uid` (with a lowercase letter)? I think that's the canonical RFC name of the attribute.

Reply:
it does. committed in https://github.com/lsst-sqre/gafaelfawr/pull/410/commits/279971917f7e2a89773798c864192a06ad718282